hazelcast/src/main/java/com/hazelcast/config/ClasspathYamlConfig.java

```diff
@@ -0,0 +1,101 @@
+     */
+    public ClasspathYamlConfig(String resource) {
+        this(resource, System.getProperties());
+    }
+
+    /**
+     * Creates a config which is loaded from a classpath resource using the
+     * Thread.currentThread contextClassLoader.
+     *
+     * @param resource   the resource, an YAML configuration file from the classpath
+     * @param properties the Properties to resolve variables in the YAML
+     * @throws IllegalArgumentException              if the resource could not be found or if properties is {@code null}
+     * @throws com.hazelcast.core.HazelcastException if the YAML content is invalid
+     */
+    public ClasspathYamlConfig(String resource, Properties properties) {
+        this(Thread.currentThread().getContextClassLoader(), resource, properties);
+    }
+
+    /**
+     * Creates a config which is loaded from a classpath resource. The System.properties are used to
+     * resolve variables in the YAML.
+     *
+     * @param classLoader the ClassLoader used to load the resource
+     * @param resource    the resource, an YAML configuration file from the classpath
+     * @throws IllegalArgumentException              if classLoader or resource is {@code null}, or if the resource is not found
+     * @throws com.hazelcast.core.HazelcastException if the YAML content is invalid
+     */
+    public ClasspathYamlConfig(ClassLoader classLoader, String resource) {
+        this(classLoader, resource, System.getProperties());
+    }
+
+    /**
+     * Creates a config which is loaded from a classpath resource.
+     *
+     * @param classLoader the ClassLoader used to load the resource
+     * @param resource    the resource, an YAML configuration file from the classpath
+     * @param properties  the properties used to resolve variables in the YAML
+     * @throws IllegalArgumentException              if classLoader or resource is {@code null}, or if the resource is not found
+     * @throws com.hazelcast.core.HazelcastException if the YAML content is invalid
+     */
+    public ClasspathYamlConfig(ClassLoader classLoader, String resource, Properties properties) {
+        if (classLoader == null) {
+            throw new IllegalArgumentException("classLoader can't be null");
+        }
+        if (resource == null) {
+            throw new IllegalArgumentException("resource can't be null");
+        }
+        if (properties == null) {
+            throw new IllegalArgumentException("properties can't be null");
+        }
+
+        LOGGER.info("Configuring Hazelcast from '" + resource + "'.");
+        InputStream in = classLoader.getResourceAsStream(resource);
+        if (in == null) {
+            throw new IllegalArgumentException("Specified resource '" + resource + "' could not be found!");
+        }
+        new YamlConfigBuilder(in).build(this);
+        //        new YamlConfigBuilder(in).setProperties(properties).build(this);
```

Comment:
Property substitution is done in the separate https://github.com/hazelcast/hazelcast/pull/14451 PR.

However, this is a very nice catch, since it is left commented out in that PR as well. Which means we have no test covering setting properties in the `Classpath***Config`, and have no test for XML counterpart either. Adding tests both for XML and YAML to the other referenced PR and leaving this as is in this PR since the proper tests can be written there. Will link it here, thanks for catching.

Reply:
Added with https://github.com/hazelcast/hazelcast/pull/14451/commits/c74c2bf16df553b00e56582cb8a02b92a2315918